corehq/util/datadog/lockmeter.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import time
+
+from corehq.util.datadog.gauges import datadog_counter, datadog_bucket_timer
+
+
+class LockMeter(object):
```

Comment:
Nit: Took me a sec to understand when I looked at the first usage; I might rename this `MeteredLock` to emphasize that in addition to measuring lock properties, it itself acts as a lock (through duck typing).

Reply:
https://github.com/dimagi/commcare-hq/pull/23428/commits/1a40483ac0424c2318a2809f64eb031b480d5b9a - I thought the name `LockMeter` had a nice ring to it, but I should have listened to that tiny voice in the back of my head that said this might not pass code review.